core/src/main/java/org/mule/transport/AbstractMessageDispatcher.java

```diff
@@ -213,4 +255,88 @@ protected void applyResponseTransformers(MuleEvent event) throws MuleException
     protected abstract void doDispatch(MuleEvent event) throws Exception;
 
     protected abstract MuleMessage doSend(MuleEvent event) throws Exception;
+
+    protected void doSendNonBlocking(MuleEvent event, CompletionHandler<MuleMessage, Exception> completionHandler)
+    {
+        throw new IllegalStateException("This MessageDispatcher does not support non-blocking");
+    }
+
+    private class NonBlockingSendCompletionHandler implements CompletionHandler<MuleMessage, Exception>
```

Comment:
I'd prefer to maintain this a private inner class.  I'll look at adding some more test tests to cover more of this though..